### ISSUE TITLE
Regularly spaced temporal axis?

### ISSUE THREAD

**Issue by petergarnaes** [Reading-eScience-Centre/coveragejson#93]
I work for the Danish Meteorological Institude (DMI), and I am exploring this standard for potential use in our Open Data services. For our climate data, we work with gridded hourly coverage data spanning many years, which creates quite a lot of points along the time axis.

As an example, we have hourly values since 2011, which is over 87600 values, and still growing. For those datetimes to be listed as an array of values in the [axis object](https://covjson.org/spec/#611-axis-objects) seems impractical.

As far as I can tell, regularly spaced temporal axis values are not supported right now. Are there any plans of supporting any type of regularly spaced temporal axis in the standard?

**Comment by petergarnaes** [comment]
Hi @jonblower, thanks for the reply! 

> Do you have a preference on any of the above?

Option 2 seems best to me. Since temporal axes are already supported as a reference system, providing some sort of temporal spacing specification seems logical (to me).

While option 1 is certainly usable, option 2 have the added benefit of being able to specify the start of the timespan, making the coverageJSON definition more descriptive and self contained in terms of external documentation.

> Maybe for time axes it would also be nicer to use a spacing (e.g. "1H") instead of num (as you know in advance the data are always going to be hourly).

I very much agree

**Comment by jonblower** [comment]
> If you have a coverage with a lot of parameters, where only some parameters are missing for a certain day, splitting the time axis wouldn't be feasible.

That's right. All the parameters in a coverage must share the same domain. The way to deal with it might be to use `null`s in the range as you suggest. It's probably not a common case though. My guess is that if a timestep is missing, it's probably because it's an observing system that has temporarily failed for some reason, and has not recorded any data for that time.

The question about open time intervals is interesting. My first instinct is to say that CoverageJSON should always record what is actually there at a given time, not what could be there in future, so a CovJSON server could continually move the end of the time interval (explicit is better than implicit). But I must admit I haven't thought about it much so there are probably valid counterarguments too.
